index.bs

```diff
@@ -0,0 +1,512 @@
+The {{hasExternalDisplay}} attribute MUST return whether the {{VRDisplay}} is separate from the device's primary display. If presenting VR content will obscure other content on the device, this should be false. When  false, the application should not attempt to mirror VR content or update non-VR UI because that content will not be visible.
+
+<dfn attribute for="VRLayer">canPresent</dfn>
+The {{canPresent}} attribute MUST return whether the {{VRDisplay}} is capable of presenting content to an HMD or similar device. Can be used to indicate "magic window" devices that are capable of 6DoF tracking but for which <codee>VRDisplay.requestPresent</code> is not meaningful. If false then calls to <codee>VRDisplay.requestPresent</code> should always fail, and <code>VRDisplay.getEyeParameters</code> should return null.
+
+
+## VREye ## {#vreye}
+
+<pre class="idl">
+enum VREye {
+  "left",
+  "right"
+};
+</pre>
+
+## VRFieldOfView ## {#vrfieldofview}
+
+The {{VRFieldOfView}} interface represents a field of view, as given by 4 degrees describing the view from a center point.
+
+<pre class="idl">
+interface VRFieldOfViewReadOnly {
+  readonly attribute double upDegrees;
+  readonly attribute double rightDegrees;
+  readonly attribute double downDegrees;
+  readonly attribute double leftDegrees;
+};
+
+interface VRFieldOfView : VRFieldOfViewReadOnly {
+  inherit attribute double upDegrees;
+  inherit attribute double rightDegrees;
+  inherit attribute double downDegrees;
+  inherit attribute double leftDegrees;
+};
+
+dictionary VRFieldOfViewInit {
+  double upDegrees = 0.0;
+  double rightDegrees = 0.0;
+  double downDegrees = 0.0;
+  double leftDegrees = 0.0;
+};
+</pre>
+
+<div class="example">
+The following code snippet creates a WebGL-compatible projection matrix from a
+{{VRFieldOfView}}.
+
+<pre class="lang-js">
+function fieldOfViewToProjectionMatrix (fov, zNear, zFar) {
+  var upTan = Math.tan(fov.upDegrees * Math.PI / 180.0);
+  var downTan = Math.tan(fov.downDegrees * Math.PI / 180.0);
+  var leftTan = Math.tan(fov.leftDegrees * Math.PI / 180.0);
+  var rightTan = Math.tan(fov.rightDegrees * Math.PI / 180.0);
+  var xScale = 2.0 / (leftTan + rightTan);
+  var yScale = 2.0 / (upTan + downTan);
+
+  var out = new Float32Array(16);
+  out[0] = xScale;
+  out[1] = 0.0;
+  out[2] = 0.0;
+  out[3] = 0.0;
+  out[4] = 0.0;
+  out[5] = yScale;
+  out[6] = 0.0;
+  out[7] = 0.0;
+  out[8] = -((leftTan - rightTan) * xScale * 0.5);
+  out[9] = ((upTan - downTan) * yScale * 0.5);
+  out[10] = -(zNear + zFar) / (zFar - zNear);
+  out[11] = -1.0;
+  out[12] = 0.0;
+  out[13] = 0.0;
+  out[14] = -(2.0 * zFar * zNear) / (zFar - zNear);
+  out[15] = 0.0;
+
+  return out;
+}
+</pre>
+</div>
+
+## VRPose ## {#vrpose}
+
+The VRPose interface represents a sensor's state at a given timestamp.
+
+<pre class="idl">
+interface VRPose {
+  readonly attribute DOMHighResTimeStamp timestamp;
+
+  readonly attribute unsigned long frameID;
```

Comment:
Is `frameID` only significant when you call `getPose`? Seems like it's hard to give a meaningful value with `getImmediatePose`. And to be honest, I'm not sure what the use case is for this value anyway.